macros/shared/src/main/scala/zio/prelude/Assertion.scala

```diff
@@ -122,7 +126,7 @@ object Assertion {
       } else (!left || !right).apply(a, negated = false)
   }
 
-  private[prelude] case class Or[A](left: Assertion[A], right: Assertion[A]) extends Assertion[A] {
+  private[prelude] final case class Or[A](left: Assertion[A], right: Assertion[A]) extends Assertion[A] {
```

Comment:
There's not much point to declaring `final` AFAIK. Especially when this is private to `prelude`.

Reply:
AFAIK Hotspot is smart enough to speculate that the case classes aren't subclassed and do the optimizations anyway. And given that they indeed aren't in practice, it won't ever have to deoptimize.

But I don't really mind :)